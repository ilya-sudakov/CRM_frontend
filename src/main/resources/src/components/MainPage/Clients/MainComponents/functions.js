import { exportClientsEmailsCSV } from '../../../../utils/xlsxFunctions.jsx'
import { getClients } from '../../../../utils/RequestsAPI/Clients.jsx'

export const sortClients = (clients, searchQuery, sortOrder) => {
  return clients.sort((a, b) => {
    if (searchQuery !== '') {
      if (sortOrder.curSort === 'nextDateContact') {
        if (new Date(a[sortOrder.curSort]) < new Date(b[sortOrder.curSort])) {
          return sortOrder[sortOrder.curSort] === 'desc' ? 1 : -1
        }
        if (new Date(a[sortOrder.curSort]) > new Date(b[sortOrder.curSort])) {
          return sortOrder[sortOrder.curSort] === 'desc' ? -1 : 1
        }
        return 0
      } else {
        if (a[sortOrder.curSort] < b[sortOrder.curSort]) {
          return sortOrder[sortOrder.curSort] === 'desc' ? 1 : -1
        }
        if (a[sortOrder.curSort] > b[sortOrder.curSort]) {
          return sortOrder[sortOrder.curSort] === 'desc' ? -1 : 1
        }
        return 0
      }
    }
  })
}

export const getEmailsExcel = () => {
  let totalClients = 1
  let clients = []
  return getClients(1)
    .then((res) => res.json())
    .then((res) => {
      totalClients = res.totalElements
      return getClients(totalClients)
    })
    .then((res) => res.json())
    .then((res) => {
      clients = res.content
      console.log(clients)
      exportClientsEmailsCSV(clients)
    })
    .catch((error) => {
      console.log(error)
    })
}
