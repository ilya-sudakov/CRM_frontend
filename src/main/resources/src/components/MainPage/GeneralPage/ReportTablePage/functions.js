export const getDaysArray = (start, end) => {
  let arr = [];
  for (let dt = new Date(start); dt <= end; dt.setDate(dt.getDate() + 1)) {
    arr.push(new Date(dt));
  }
  return arr;
};

export const sortEmployees = (employees) => {
  return employees.sort((a, b) => {
    if (a.employee.lastName < b.employee.lastName) {
      return -1;
    }
    if (a.employee.lastName > b.employee.lastName) {
      return 1;
    }
    return 0;
  });
};

export const getMonthDates = (curDate) => {
  const date =
    new Date(curDate.getFullYear(), curDate.getMonth() + 1, 0).getDate() + 1;
  //Получаем массив с датами месяца
  let dates = [];
  for (let i = 1; i < date; i++) dates.push(i);

  return dates;
};
