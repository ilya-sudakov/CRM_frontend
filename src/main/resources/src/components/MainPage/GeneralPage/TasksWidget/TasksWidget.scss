.tasks-widget {
    font-size: 1rem;
    &__content {
        width: 100%;
    }
    &__list {
        width: 100%;
        .tasks-widget__list-item {
            min-width: 100%;
            padding: 5px 0px;
            box-sizing: border-box;
            margin-bottom: 5px;
        }
    }
}
