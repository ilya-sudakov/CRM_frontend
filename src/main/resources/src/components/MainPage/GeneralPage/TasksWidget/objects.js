export const conditions = {
  ['Выполнено']: 'completed',
  ['Материалы']: 'materials',
  ['В процессе']: 'in-progress',
  ['Отложено']: 'delayed',
  ['Проблема']: 'problem',
}
