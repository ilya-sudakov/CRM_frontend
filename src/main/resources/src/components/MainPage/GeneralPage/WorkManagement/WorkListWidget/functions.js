export const sortEmployeesObject = (employees) => {
  return employees.sort((a, b) => {
    if (a[1].lastName < b[1].lastName) {
      return -1;
    }
    if (a[1].lastName > b[1].lastName) {
      return 1;
    }
    return 0;
  });
};

export const filterEmployeesObject = (employees, workshop) => {
  return employees.filter((employee) => {
    const item = employee[1];
    if (item.workshop === workshop.name) return true;
  });
};

export const getAllEmployees = (works) => {
  let newEmployees = {};
  works.map((work) => {
    if (newEmployees[work.employee.id] === undefined) {
      return (newEmployees = Object.assign({
        ...newEmployees,
        [work.employee.id]: work.employee,
      }));
    }
  });
  return newEmployees;
};
