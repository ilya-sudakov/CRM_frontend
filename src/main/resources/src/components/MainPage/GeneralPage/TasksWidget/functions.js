import { getMainTasks } from '../../../../utils/RequestsAPI/MainTasks.jsx'
import { formatDateStringNoYear } from '../../../../utils/functions.jsx'

export const filterTasks = (tasks) => {
  return tasks.filter((task) => task.condition !== 'Выполнено')
}

export const filterTasksByUser = (tasks, userRole = 'admin') => {
  return tasks.filter((task) => task.responsible === userRole)
}

export const getTasksList = async () => {
  let tasks = []
  await getMainTasks()
    .then((res) => res.json())
    .then((res) => {
      return (tasks = res)
    })
    .catch((err) => {
      return console.error(err)
    })
  return tasks
}

export const getTasksControlDatesList = (tasks) => {
  let controlDates = {}
  tasks
    .sort((a, b) => {
      if (a.dateControl < b.dateControl) {
        return -1
      }
      if (a.dateControl > b.dateControl) {
        return 1
      }
      return 0
    })
    .map((task) => {
      const curDate = formatDateStringNoYear(task.dateControl)

      return (controlDates = {
        ...controlDates,
        [curDate]:
          controlDates[curDate] === undefined
            ? [task]
            : [...controlDates[curDate], task],
      })
    })
  return controlDates
}
