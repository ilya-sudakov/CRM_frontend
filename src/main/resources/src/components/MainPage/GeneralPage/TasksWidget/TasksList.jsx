import React from 'react'
import PlaceholderLoading from '../../../../utils/TableView/PlaceholderLoading/PlaceholderLoading.jsx'
import {
  formatDateString,
  formatDateStringNoYear,
  dateDiffInDays,
} from '../../../../utils/functions.jsx'
import { conditions } from './objects.js'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

const TasksList = ({ tasks, isLoading, controlDates }) => {
  return (
    <div className="tasks-widget__list">
      {isLoading ? (
        <PlaceholderLoading
          itemClassName="list__item list__item--placeholder"
          minHeight="2rem"
          items={3}
        />
      ) : (
        Object.entries(controlDates).map((date, index) => {
          const isExpired = new Date(date[0]) < new Date()
          return (
            <ListWrapper
              isExpired={isExpired}
              index={index}
              date={date}
              tasks={tasks}
            />
          )
        })
      )}
    </div>
  )
}

export default TasksList

TasksList.propTypes = {
  tasks: PropTypes.array,
  isLoading: PropTypes.bool,
  controlDates: PropTypes.object,
}

const ListWrapper = ({ isExpired, index, date, tasks }) => {
  return (
    <div
      className={`tasks-widget__date-wrapper ${
        isExpired ? 'tasks-widget__date-wrapper--expired' : ''
      }`}
      key={index}
    >
      <div
        className={`tasks-widget__date ${
          isExpired ? 'tasks-widget__date--expired' : ''
        }`}
      >{`до ${formatDateStringNoYear(date[0])} ${
        isExpired
          ? `- опоздание ${dateDiffInDays(new Date(date[0]), new Date())} дн.`
          : ''
      }`}</div>
      {tasks
        .filter(
          (task) =>
            formatDateString(task.dateControl) === formatDateString(date[0]),
        )
        .map((task) => (
          <ListItem task={task} />
        ))}
    </div>
  )
}

ListWrapper.propTypes = {
  tasks: PropTypes.array,
  index: PropTypes.number,
  isExpired: PropTypes.bool,
  date: PropTypes.object,
}

const ListItem = ({ task }) => {
  return (
    <div className={`list__item list__item--${conditions[task.condition]}`}>
      <Link
        className="list-item__general-info"
        to={`/dispatcher/general-tasks/edit/${task.id}`}
      >
        <span className="list-item__description">{task.description}</span>
        <span className="list-item__conditions">
          <span className="condition condition--status">{task.condition}</span>
          <span
            className={`condition condition--date ${
              task.status === '' || task.status === null
                ? 'condition--hidden'
                : ''
            }`}
          >{`от ${formatDateStringNoYear(task.dateCreated)}`}</span>
          <span className="condition condition--description">
            {task.status}
          </span>
        </span>
      </Link>
    </div>
  )
}

ListItem.propTypes = {
  task: PropTypes.Object,
}
