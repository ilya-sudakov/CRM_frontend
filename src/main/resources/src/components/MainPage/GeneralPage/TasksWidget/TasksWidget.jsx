import React from 'react'
import Widget from '../Widget/Widget.jsx'
import './TasksWidget.scss'

const TasksWidget = () => {
  return (
    <div>
      <Widget title="Задачи" />
    </div>
  )
}

export default TasksWidget
