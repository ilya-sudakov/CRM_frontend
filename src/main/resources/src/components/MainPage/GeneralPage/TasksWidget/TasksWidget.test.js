import React from "react";
import TasksWidget from "./TasksWidget.jsx";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import { renderWithRouterAndContext } from "../../../../utils/testing/functions.js";
import {
  filterTasks,
  filterTasksByUser,
  getTasksControlDatesList,
  getTasksList,
} from "./functions.js";
import "isomorphic-fetch";

afterEach(cleanup);

describe("TasksWidget component", () => {
  //COMPONENT
  it("renders", () => {
    renderWithRouterAndContext(<TasksWidget />);
  });

  it("matches snapshot", () => {
    const { asFragment } = renderWithRouterAndContext(<TasksWidget />);
    expect(asFragment()).toMatchSnapshot();
  });

  // FUNCTIONS
  it("filters tasks", () => {
    const tasks = [{ condition: "Выполнено" }, { condition: "123" }];
    const expectedTasks = [{ condition: "123" }];
    expect(filterTasks(tasks)).toEqual(expectedTasks);
  });

  it("filters tasks by User", () => {
    const tasks = [
      { condition: "Выполнено" },
      { condition: "123", responsible: "admin" },
    ];
    const expectedTasks = [{ condition: "123", responsible: "admin" }];
    expect(filterTasksByUser(tasks)).toEqual(expectedTasks);
  });

  it("fetches tasks list", () => {
    expect(getTasksList());
  });

  it("getTasksControlDates", () => {
    const tasks = [
      { condition: "Выполнено", dateControl: new Date(2012, 1, 1) },
      {
        condition: "123",
        responsible: "admin",
        dateControl: new Date(2012, 1, 1),
      },
    ];
    const expectedDates = {
      "Wed Feb 01 2012 00:00:00 GMT+0400 (GMT+03:00)": [
        { condition: "Выполнено", dateControl: new Date(2012, 1, 1) },
        {
          condition: "123",
          responsible: "admin",
          dateControl: new Date(2012, 1, 1),
        },
      ],
    };
    expect(getTasksControlDatesList(tasks)).toEqual(expectedDates);
  });
});
