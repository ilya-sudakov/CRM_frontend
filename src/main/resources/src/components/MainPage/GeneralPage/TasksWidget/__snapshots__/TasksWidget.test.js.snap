// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksWidget component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="authorization"
  >
    <div
      class="authorization__background"
    >
      <div
        class="authorization__circle authorization__circle--1"
      />
      <div
        class="authorization__circle authorization__circle--2"
      />
      <div
        class="authorization__circle authorization__circle--3"
      />
    </div>
    <div
      class="main-window"
    >
      <div
        class="authorization__panel"
      >
        <div
          class="authorization__title"
        >
          <img
            alt=""
            class="authorization__img authorization__img--logo"
            src="[object Object]"
          />
        </div>
        <div
          class="window_error window_error--hidden"
        >
          <div
            class="window_error__content window_error__content--hidden"
          >
            <div
              class="window_error__title"
            >
              Ошибка!
              <div
                class="window_error__exit"
              >
                <div
                  class="window_error__bar"
                />
                <div
                  class="window_error__bar"
                />
              </div>
            </div>
            <div
              class="window_error__message"
            >
              Ошибка при авторизации
            </div>
            <div
              class="main-window__button"
            >
              ОК
            </div>
          </div>
        </div>
        <div
          class="authorization__field_input"
        >
          <div
            class="authorization__field_name "
          >
            Логин
          </div>
          <input
            autocomplete="on"
            id="login"
            type="text"
            value=""
          />
        </div>
        <div
          class="authorization__field_input"
        >
          <div
            class="authorization__field_name  "
          >
            Пароль
          </div>
          <img
            alt=""
            class="authorization__img authorization__img--eye"
            src="[object Object]"
          />
          <div
            class="authorization__line authorization__line--hidden"
          />
          <input
            autocomplete="on"
            id="password"
            type="password"
          />
        </div>
        <div
          class="checkbox"
        >
          <label
            class="checkbox__container"
          >
            <input
              checked=""
              name=""
              type="checkbox"
              value=""
            />
            <div
              class="checkbox__checkmark"
            />
            <span
              class="checkbox__text"
            >
              Запомнить меня
            </span>
          </label>
        </div>
        <button
          class="button  main-window__button "
          type="submit"
        >
          <img
            alt=""
            class="button__img "
            src="[object Object]"
          />
          <span
            class="button__text "
          >
            Войти
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
